### PR TITLE
Add bh1900nux driver

### PR DESCRIPTION
**Pull request by wosym**
### Contribution description
Adds a driver for the BH1900NUX temperature sensor. 
I used the BH1750FVI driver by @haukepetersen as a reference so that drivers for sensors of the same family would have the same structure. 

### Testing procedure
- Connect the sensor to your device
- Change default parameters if they differ from yours
- Run test app

### CODE REVIEW

File: drivers/bh1900nux/bh1900nux.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    /* initialize the device descriptor */
+    dev->i2c = params->i2c;
+    dev->addr = params->addr;
+
+    return BH1900NUX_OK;
+}
+
+int bh1900nux_read(const bh1900nux_t *dev, double *temp)
```
Comment (benpicco) [inline drivers/bh1900nux/bh1900nux.c]: Don't use floats if they can be avoided.
The device has a resolution of 1/16 °C, so just return mC here if you don't want to return the raw value as (16 * °C)

File: drivers/bh1900nux/bh1900nux.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    ret = i2c_read_regs(dev->i2c, dev->addr, BH1900NUX_REG_ADDR, bytes, 2, 0);
+    if(ret < 0) {
+        return ret;
+    }
+    i2c_release(dev->i2c);
+
+    /* Calculate temperature */
+    tmp = ((uint16_t)bytes[0] << 8) | bytes[1];
+    *temp = 0.0625 * ((double) tmp / 16);
```
Comment (benpicco) [inline drivers/bh1900nux/bh1900nux.c]: Shouldn't dividing by 16 once be enough? 😉 
You can avoid the floating point arithmetic entirely by multiplying by 1000 first and then divide by 16, returning mC.

File: drivers/bh1900nux/bh1900nux.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    int tmp = 0;
+    uint8_t bytes[2] = {0};
+
+    /* Read raw sensor value */
+    DEBUG("[bh1900nux] read temperature\n");
+    ret = i2c_acquire(dev->i2c);
+    if(ret < 0) {
+        return BH1900NUX_ERR_I2C;
+    }
+    ret = i2c_read_regs(dev->i2c, dev->addr, BH1900NUX_REG_ADDR, bytes, 2, 0);
```
Comment (benpicco) [inline drivers/bh1900nux/bh1900nux.c]:
```suggestion
    uint16_t raw;
    ret = i2c_read_regs(dev->i2c, dev->addr, BH1900NUX_REG_ADDR, &raw, sizeof(raw), 0);
    raw = ntohs(raw);
```

Comment (benpicco) [reply drivers/bh1900nux/bh1900nux.c]: You will also also have to modify the rest of the function, this was just a draft 😉

File: drivers/bh1900nux/bh1900nux.c (hunk 2 above)
Comment (benpicco) [reply drivers/bh1900nux/bh1900nux.c]: Now I get the data format.
Forget about the `ntohs()`:

`bytes[0]` has the whole °C - you can use this as is, or multiply it by 1000 for mC.
`bytes[1]` has the fractional part * 16 - just shift right by 4 places to get the fractional part in mC.

File: drivers/bh1900nux/bh1900nux.c [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    ret = i2c_read_regs(dev->i2c, dev->addr, BH1900NUX_REG_ADDR, &raw, sizeof(raw), 0);
+    raw = ntohs(raw);
+    if(ret < 0) {
+        return ret;
+    }
+    i2c_release(dev->i2c);
+
+    /* Calculate temperature */
+    *temp  = raw * 3.3333;
```
Comment (benpicco) [inline drivers/bh1900nux/bh1900nux.c]:
```suggestion
    *temp  = (raw * 1000) / 256;
```

File: drivers/bh1900nux/bh1900nux.c (hunk 2 above)
Comment (wosym) [reply drivers/bh1900nux/bh1900nux.c]: Resolved the first comment, second part is still up for debate (discussed on IRC).

File: drivers/include/bh1900nux.h [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+ * The result value is the measured temperature in MILICELSUIS and ranges
+ * from -30 to 95.
+ *
+ * @param[in] dev       device descriptor of the targeted device
+ * @param[out] temp     temperature output
+ *
+ * @return  BH1900NUX_OK on succes
+ * @return  BH1900NUX_ERR_I2C on failure
+ */
+int bh1900nux_read(const bh1900nux_t *dev, uint16_t *temp);
```
Comment (benpicco) [inline drivers/include/bh1900nux.h]:
```suggestion
int bh1900nux_read(const bh1900nux_t *dev,  int16_t *temp);
```

File: drivers/bh1900nux/bh1900nux.c [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    dev->i2c = params->i2c;
+    dev->addr = params->addr;
+
+    return BH1900NUX_OK;
+}
+
+int bh1900nux_read(const bh1900nux_t *dev, uint16_t *temp)
+{
+    int ret = 0;
+    uint16_t raw;
```
Comment (benpicco) [inline drivers/bh1900nux/bh1900nux.c]:
```suggestion
    int16_t raw;
```
> Negative numbers are represented in binary twos complement format.

File: drivers/bh1900nux/bh1900nux.c [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    DEBUG("[bh1900nux] read temperature\n");
+    ret = i2c_acquire(dev->i2c);
+    if (ret < 0) {
+        return BH1900NUX_ERR_I2C;
+    }
+    ret =
+        i2c_read_regs(dev->i2c, dev->addr, BH1900NUX_REG_ADDR, &raw,
+                      sizeof(raw),
+                      0);
+    raw = ntohs(raw);
```
Comment (benpicco) [inline drivers/bh1900nux/bh1900nux.c]:
```suggestion
    raw = ntohs(raw) >> 4;
```
Let's get rid of those 0 bytes, they will just add confusion for negative numbers (two's complement)

File: drivers/bh1900nux/bh1900nux.c [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    if (ret < 0) {
+        return ret;
+    }
+    i2c_release(dev->i2c);
+
+    /* Calculate temperature */
+    *temp = (raw * 1000) / 256;
```
Comment (benpicco) [inline drivers/bh1900nux/bh1900nux.c]:
```suggestion
    *temp = ((int) raw * 1000) / 16;
```
just to be sure

File: drivers/bh1900nux/bh1900nux.c [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    uint16_t raw;
+
+    /* Read raw sensor value */
+    DEBUG("[bh1900nux] read temperature\n");
+    ret = i2c_acquire(dev->i2c);
+    if (ret < 0) {
+        return BH1900NUX_ERR_I2C;
+    }
+    ret =
+        i2c_read_regs(dev->i2c, dev->addr, BH1900NUX_REG_ADDR, &raw,
```
Comment (benpicco) [inline drivers/bh1900nux/bh1900nux.c]: `uncrustify` was a bit overeager here, this can stay on one line.

**Comment by benpicco** [comment]
[Travis](https://travis-ci.org/RIOT-OS/RIOT/builds/648801726) found some issues.

**Comment by wosym** [comment]
@benpicco : passed now :)

File: drivers/include/bh1900nux.h [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    Possible I2C bus addresses of the device
+ *
+ * The actual address of the device depends on the states of the slave adress pins.
```
Comment (benpicco) [inline drivers/include/bh1900nux.h]:
```suggestion
 * The actual address of the device depends on the states of the slave address pins.
```

File: drivers/include/bh1900nux.h [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+/**
+ * @brief   Read the temperature measerd by the device [MILICELSIUS].
+ *
+ * The result value is the measured temperature in MILICELSUIS and ranges
+ * from -30 to 95.
+ *
+ * @param[in] dev       device descriptor of the targeted device
+ * @param[out] temp     temperature output
+ *
+ * @return  BH1900NUX_OK on succes
```
Comment (benpicco) [inline drivers/include/bh1900nux.h]:
```suggestion
 * @return  BH1900NUX_OK on success
```

File: drivers/bh1900nux/bh1900nux.c [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    uint16_t raw;
+
+    /* Read raw sensor value */
+    DEBUG("[bh1900nux] read temperature\n");
+    ret = i2c_acquire(dev->i2c);
+    if (ret < 0) {
+        return BH1900NUX_ERR_I2C;
+    }
+    ret = i2c_read_regs(dev->i2c, dev->addr, BH1900NUX_REG_ADDR, &raw, sizeof(raw), 0);
+    raw = ntohs(raw) >> 4;
```
Comment (benpicco) [inline drivers/bh1900nux/bh1900nux.c]: I'd move that down to the `/* Calculate temperature */` section.

File: drivers/bh1900nux/bh1900nux.c [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+    if (ret < 0) {
+        return ret;
+    }
+    i2c_release(dev->i2c);
+
+    /* Calculate temperature */
+    *temp = ((int32_t) raw * 1000) / 16;
+
+
```
Comment (benpicco) [inline drivers/bh1900nux/bh1900nux.c]: drop the extra newline.

**Comment by wosym** [comment]
@benpicco wouldn't hurt to have someone confirm if the temperatures are right. They seem a little high to be honest. But that might also be because of sensor inaccuracy.

File: drivers/bh1900nux/bh1900nux.c [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    dev->i2c = params->i2c;
+    dev->addr = params->addr;
+
+    return BH1900NUX_OK;
+}
+
+int bh1900nux_read(const bh1900nux_t *dev, int16_t *temp)
+{
+    int ret = 0;
+    uint16_t raw;
```
Comment (benpicco) [inline drivers/bh1900nux/bh1900nux.c]:
```suggestion
    int16_t raw;
```

File: drivers/bh1900nux/bh1900nux.c [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+        return BH1900NUX_ERR_I2C;
+    }
+    ret = i2c_read_regs(dev->i2c, dev->addr, BH1900NUX_REG_ADDR, &raw, sizeof(raw), 0);
+    if (ret < 0) {
+        return ret;
+    }
+    i2c_release(dev->i2c);
+
+    /* Calculate temperature */
+    raw = ntohs(raw) >> 4;
```
Comment (benpicco) [inline drivers/bh1900nux/bh1900nux.c]:
```suggestion
    raw = (int16_t)ntohs(raw) >> 4;
```

**Pull request merged**